### PR TITLE
refactor: Remove duplication logic in start_span

### PR DESCRIPTION
**Pull request by miry**
There are 2 places in sdk, where checked mostly the same code parent_span_id and
trace_id base on parent_span's context.

Because there is no other places in gems,
that uses method `internal_start_span`, I left only logic 'internal_start_span' and removed unused attribute.

It is possible opposite: to have checking of context in `start_span`.

### CODE REVIEW

**Comment by miry** [comment]
I don't think the failed checks related to my changes. I could not re run checks as well.

**Comment by ahayworth** [comment]
> I don't think the failed checks related to my changes. I could not re run checks as well.

@miry I have kicked off the failed jobs again - if you see this option, you can do so as well in the future:

Screen Shot 2022-05-24 at 4 32 18 PM

(I'm not sure if that's something available to all contributors, or just members of the project specifically)

File: sdk/lib/opentelemetry/sdk/trace/tracer.rb
Comment (fbogsany) [inline sdk/lib/opentelemetry/sdk/trace/tracer.rb:12]: There's no real value in extracting this named constant - it is only used in one place, and this makes it *less* readable.

**Review by fbogsany — CHANGES_REQUESTED** [review]
I'm going to block this until I can read the history closely. The change that introduced `internal_create_span` was done very carefully, so I'm surprised this duplication slipped through. I suspect some subsequent change introduced this (I'm particularly surprised by the repeated calls to `Trace.current_span(with_parent)`).

**Comment by fbogsany** [comment]
https://github.com/open-telemetry/opentelemetry-ruby/pull/782 is the culprit that introduced the duplication.

https://github.com/open-telemetry/opentelemetry-ruby/pull/822 is the original refactor.

**Comment by fbogsany** [comment]
I'll stare at this some more tonight/tomorrow.

**Comment by miry** [comment]
> (I'm not sure if that's something available to all contributors, or just members of the project specifically)

Yep. it is only available for maintainers.

File: sdk/lib/opentelemetry/sdk/trace/tracer.rb
Comment (miry) [reply sdk/lib/opentelemetry/sdk/trace/tracer.rb]: reverted this change

**Review by fbogsany — COMMENTED** [review]
You can simplify this further by pushing the `parent_span =` line into `internal_create_span` and removing that parameter. I think we should do the "input validation" in `start_span` and remove it from `internal_create_span`. I can't comment on those lines, but that means removing the `name ||= 'empty'` and `kind ||= :internal` lines from `internal_create_span`.

File: sdk/lib/opentelemetry/sdk/trace/tracer_provider.rb
Comment (miry) [reply sdk/lib/opentelemetry/sdk/trace/tracer_provider.rb]: applied in https://github.com/open-telemetry/opentelemetry-ruby/pull/1281/commits/074f6fdfb0e65f3634639519d05f60f5017d0585

File: sdk/lib/opentelemetry/sdk/trace/tracer.rb
Comment (miry) [reply sdk/lib/opentelemetry/sdk/trace/tracer.rb]: applied in https://github.com/open-telemetry/opentelemetry-ruby/pull/1281/commits/074f6fdfb0e65f3634639519d05f60f5017d0585

File: sdk/lib/opentelemetry/sdk/trace/tracer.rb
Comment (miry) [reply sdk/lib/opentelemetry/sdk/trace/tracer.rb]: applied in https://github.com/open-telemetry/opentelemetry-ruby/pull/1281/commits/074f6fdfb0e65f3634639519d05f60f5017d0585

**Comment by miry** [comment]
@fbogsany Good cacth. I applied all your changes and pushed.

**Review by fbogsany — APPROVED** [review]

**Pull request merged**
